src/Http/Http.Extensions/src/ProblemDetailsDefaultWriter.cs

```diff
@@ -0,0 +1,63 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text.Json.Serialization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.Http;
+
+internal sealed partial class ProblemDetailsDefaultWriter : IProblemDetailsWriter
+{
+    private static readonly MediaTypeHeaderValue _jsonMediaType = new("application/json");
+    private static readonly MediaTypeHeaderValue _problemDetailsJsonMediaType = new("application/problem+json");
+    private readonly ProblemDetailsOptions _options;
+
+    public ProblemDetailsDefaultWriter(IOptions<ProblemDetailsOptions> options)
+    {
+        _options = options.Value;
+    }
+
+    [UnconditionalSuppressMessage("Trimming", "IL2026",
+        Justification = "JSON serialization of ProblemDetails.Extensions might require types that cannot be statically analyzed and we need to fallback" +
+        "to reflection-based. The ProblemDetailsConverter is marked as RequiresUnreferencedCode already.")]
+    public async ValueTask<bool> TryWriteAsync(ProblemDetailsContext context)
+    {
+        var httpContext = context.HttpContext;
+        var acceptHeader = httpContext.Request.Headers.Accept.GetList<MediaTypeHeaderValue>();
+
+        if (acceptHeader == null ||
+            !acceptHeader.Any(h => _jsonMediaType.IsSubsetOf(h) || _problemDetailsJsonMediaType.IsSubsetOf(h)))
+        {
+            return false;
+        }
+
+        ProblemDetailsDefaults.Apply(context.ProblemDetails, httpContext.Response.StatusCode);
+        _options.CustomizeProblemDetails?.Invoke(context);
+
+        if (context.ProblemDetails.Extensions is { Count: 0 })
```

Comment:
Since `ProblemDetails.Extensions` causes issues during the serialization using JsonSerializerContext, maybe because of this https://github.com/dotnet/runtime/issues/71714, I decided to use the source generation only when no extensions are defined. 

I am ok to remove this and always use the reflection-based if it is a better approach.